### DISCUSSION TITLE
Design Decision Question

### DISCUSSION THREAD

**Discussion by cemremengu** [openobserve/openobserve#2945] [Q&A] [answered]
Curious about why you did not build your product on a db like clickhouse which is well suited for this domain with great performance and compression. 

Are there some advantages of your storage approach or is it purely preference?

**Comment by prabhatsharma** [comment, marked as answer]
Ciclkhouse is an analytics tool built for a purpose different from observability. Something built specifically for observability will be able to deal with observability requirements much better. Take two aspects where OpenObserve shines already - 

1. Single binary for logs, metrics, traces, dashboards, alerting, and more. 
2. object store as primary data storage mechanism. 

These two things make it a lot better for most people wanting a specific tool for observability. 

Additionally we do not think that clickhouse works well for logs. Logs can have very different set of columns and you can't really create columns at runtime in clickhouse. Moreso datafusion is quite fast too and has allowed us to be quite productive and has allowed us to build a much simpler solution with stateless nodes. Can't really do that with clickhouse.

**Reply by cemremengu** [reply]
Thank you so much for answering and good luck with building OpenObserve!
